src/file-events/headers/linux_fsnotifier.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#ifdef __linux__
+
+#include <poll.h>
+#include <sys/eventfd.h>
+#include <sys/inotify.h>
+#include <unordered_map>
+
+#include "generic_fsnotifier.h"
+#include "net_rubygrapefruit_platform_internal_jni_LinuxFileEventFunctions.h"
+
+using namespace std;
+
+class Server;
```

Comment:
I wonder if you only use the classes defined in this header file in the Linux watcher, shouldn't those be moved to `linux_fsnotifier.h`? Same for the other headers.

Reply:
This is in `linux_fsnotifier.h`. Do you mean to `linux_fsnotifier.cpp`?

Reply:
D'oh, yes I do.

Reply:
What would be the advantage?

Reply:
Fewer files to look at and less indirection.

Reply:
You can't decrease the indirection, as the forward declarations would still be needed, making the `.cpp` file even longer. It also seems idiomatic to have a header file, though I don't observe this with the really internal types.

Reply:
And it's a nice place to get an overview of what's there. Think of it as our separation between interface and the `Default*` implementation.